server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeFiltersTests.java

```diff
@@ -295,6 +295,16 @@ public void testOnlyAttributeValueFilter() {
         assertThat(discoveryNodeFilters.isOnlyAttributeValueFilter(), is(discoveryNodeFilters.match(node)));
     }
 
+    public void testIpv6WithScopeIdFromSettingsMatchingIpv6WithCompressedScopeId() {
+        Settings settings = shuffleSettings(Settings.builder()
+            .put("xxx." + randomFrom("_ip", "_host_ip", "_publish_ip"), "fdbd:dc00:111:222:0:0:0:333")
```

Comment:
I suppose since it's important that we don't normalise things like node names we should have a test to assert that too.